src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java

```diff
@@ -491,7 +491,13 @@ private void ensureReplicatorThread() throws Exception {
 				// standard binlog positioning is a lot easier; we can really reconnect anywhere
 				// we like, so we don't have to bail out of the middle of an event.
 				LOGGER.warn("replicator stopped at position: {} -- restarting", client.getBinlogFilename() + ":" + client.getBinlogPosition());
+
+				Long oldMasterId = client.getMasterServerId();
 				tryReconnect();
+				if (client.getMasterServerId() != oldMasterId) {
+					throw new Exception(
+						"Master id changed from " + oldMasterId + " to " + client.getMasterServerId() + " while using binlog coordinate positioning. Cannot continue with the info that we have");
+				}
```

Comment:
this should only be an exception in non-gtid mode -- otherwise a logger.warn should be fine

Reply:
Oh I think is a formatting issue, which I will correct. This is under the `else` block at 490, which if I am not mistaken re-presents non gtid mode

Reply:
Actually, it seems like github oddity. I have the settings specified for a tab size of 8 and no expand tab. However, for some reason github is showing a tab size of 4 for my changes _only_ on this PR but on the view version of the [file](https://github.com/zendesk/maxwell/blob/733249adc2c0c25ff6aee4545a0455060420ac48/src/main/java/com/zendesk/maxwell/replication/BinlogConnectorReplicator.java#L491). 

Any ideas as to why the PR has odd formatting vs the actual files?

Reply:
i do not know.  it's interesting but i have no idea what would be happening

Reply:
> This is under the else block at 490, which if I am not mistaken represents non gtid mode

ahhh you're right i'm dumb